File: eddy-audio-main/examples/cpp/whisper_example.cpp
// Copyright (C) 2025 Eddy SDK
// SPDX-License-Identifier: Apache-2.0

#include "eddy/pipelines/whisper_pipeline.hpp"

#include <iostream>
#include <iomanip>
#include <exception>

int main(int argc, char* argv[]) {
    try {
        if (argc < 3) {
            std::cerr << "Usage: " << argv[0] << " <MODEL_DIR> <WAV_FILE> [DEVICE] [LANGUAGE]" << std::endl;
            std::cerr << "Example: " << argv[0] << " ./models/whisper-v3-turbo ./audio.wav NPU en" << std::endl;
            return 1;
        }

        std::string model_path = argv[1];
        std::string wav_file = argv[2];
        std::string device = argc > 3 ? argv[3] : "NPU";
        std::string language = argc > 4 ? argv[4] : "en";

        std::cout << "=== Eddy Whisper Example ===" << std::endl;
        std::cout << "Model: " << model_path << std::endl;
        std::cout << "Audio: " << wav_file << std::endl;
        std::cout << "Device: " << device << std::endl;
        std::cout << "Language: " << language << std::endl;
        std::cout << std::endl;

        // Configure Whisper pipeline
        eddy::WhisperConfig config;
        config.model_path = model_path;
        config.device = device;
        config.language = language;
        config.task = "transcribe";
        config.return_timestamps = true;
        config.enable_cache = true;
        config.cache_dir = "./cache";

        // Create pipeline
        std::cout << "Creating Whisper pipeline..." << std::endl;
        eddy::WhisperPipeline pipeline(config);
        std::cout << std::endl;

        // Transcribe
        std::cout << "Transcribing audio..." << std::endl;
        auto result = pipeline.transcribe(wav_file);
        std::cout << std::endl;

        // Print results
        std::cout << "=== Transcription Result ===" << std::endl;
        std::cout << "Text: " << result.text << std::endl;
        std::cout << "Confidence: " << std::fixed << std::setprecision(2)
                  << (result.confidence * 100.0f) << "%" << std::endl;
        std::cout << "Inference Time: " << std::fixed << std::setprecision(2)
                  << result.inference_duration_ms << " ms" << std::endl;
        std::cout << std::endl;

        // Print timestamps if available
        if (!result.chunks.empty()) {
            std::cout << "=== Timestamps ===" << std::endl;
            for (const auto& chunk : result.chunks) {
                std::cout << "[" << std::fixed << std::setprecision(2)
                          << chunk.start_ts << " -> ";
                if (chunk.end_ts >= 0) {
                    std::cout << chunk.end_ts;
                } else {
                    std::cout << "?";
                }
                std::cout << "] " << chunk.text << std::endl;
            }
        }

        return 0;

    } catch (const std::exception& e) {
        std::cerr << "Error: " << e.what() << std::endl;
        return 1;
    }
}

File: eddy-audio-main/bindings/csharp/examples/WhisperExample.cs
// Copyright (C) 2025 Fluid Inference
// SPDX-License-Identifier: Apache-2.0

using System;
using Eddy;

class Program
{
    static void Main(string[] args)
    {
        if (args.Length < 2)
        {
            Console.WriteLine("Usage: WhisperExample <MODEL_DIR> <WAV_FILE> [DEVICE] [LANGUAGE]");
            Console.WriteLine("Example: WhisperExample ./models/whisper-v3-turbo ./audio.wav NPU en");
            return;
        }

        string modelPath = args[0];
        string wavFile = args[1];
        string device = args.Length > 2 ? args[2] : "NPU";
        string language = args.Length > 3 ? args[3] : "en";

        try
        {
            Console.WriteLine("=== eddy Whisper C# Example ===");
            Console.WriteLine($"Model: {modelPath}");
            Console.WriteLine($"Audio: {wavFile}");
            Console.WriteLine($"Device: {device}");
            Console.WriteLine($"Language: {language}");
            Console.WriteLine();

            // Configure Whisper pipeline
            var config = new WhisperConfig
            {
                ModelPath = modelPath,
                Device = device,
                Language = language,
                Task = "transcribe",
                ReturnTimestamps = true,
                EnableCache = true,
                CacheDir = "./cache"
            };

            // Create pipeline
            Console.WriteLine("Creating Whisper pipeline...");
            using var pipeline = new WhisperPipeline(config);
            Console.WriteLine();

            // Transcribe
            Console.WriteLine("Transcribing audio...");
            var result = pipeline.Transcribe(wavFile);
            Console.WriteLine();

            // Print results
            Console.WriteLine("=== Transcription Result ===");
            Console.WriteLine($"Text: {result.Text}");
            Console.WriteLine($"Confidence: {result.Confidence * 100:F2}%");
            Console.WriteLine($"Inference Time: {result.InferenceDurationMs:F2} ms");
            Console.WriteLine();

            // Print timestamps if available
            if (result.Chunks.Count > 0)
            {
                Console.WriteLine("=== Timestamps ===");
                foreach (var chunk in result.Chunks)
                {
                    string endTime = chunk.EndTime >= 0 ? $"{chunk.EndTime:F2}" : "?";
                    Console.WriteLine($"[{chunk.StartTime:F2} -> {endTime}] {chunk.Text}");
                }
            }
        }
        catch (EddyException ex)
        {
            Console.Error.WriteLine($"eddy error: {ex.Message}");
            Environment.Exit(1);
        }
        catch (Exception ex)
        {
            Console.Error.WriteLine($"Error: {ex.Message}");
            Environment.Exit(1);
        }
    }
}
